### PR TITLE
UI Improvements to Instructor Report Page

### LINKED ISSUE #2: FEATURE: UX/UI Improvements to Instructor Report page
# User Story

* As an instructor
* I can see an improved user interface on the instructor report page 
* So that I can I have an improved user experience

### Discussion

On the instructor report page, dollar amounts are not formatted as currency the way they are on the leaderboard page:

Compare:

image

vs.

image

We'd like dollar amounts formatted as currency, and we'd like other numbers right justified rather than left justified in their columns.

### Acceptance Criteria

- [ ] Dollar amounts on instructor report page are displayed as currency
- [ ] Other numbers are right justified.

### Implementation Todos

Should be a change to the react component that renders the instructor report page.

### PR DESCRIPTION
**Pull request by mitali-g**
## Overview

In this PR, we make UI improvements to the Instructor Report page. We format dollar amounts as currency and all other numbers are right justified rather than left justified in their columns.

## Screenshots

Screenshot 2023-11-21 at 12 13 13 AM

## Validation

1. Download this branch.
2. Run the project.
3. Go to the Manage Jobs page and generate an Instructor Report.
4. Go to the Instructor Report page and make sure the values are right justified and dollar amounts are formatted as currency.

## Tests

- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #2

### CODE REVIEW

File: frontend/src/main/components/Leaderboard/LeaderboardTable.js
Comment (katytsao) [inline frontend/src/main/components/Leaderboard/LeaderboardTable.js:1]: (these changes are not related to the actual pr, just formatting)

File: frontend/package-lock.json
Comment (katytsao) [inline frontend/package-lock.json:1]: i don't think this file is supposed to be edited?

File: frontend/src/tests/components/Reports/ReportHeaderTable.test.js [hunk 1]
```diff
@@ -22,7 +23,7 @@ describe("ReportHeaderTable tests", () => {
     const expectedFields = [ 'cowPrice', 'milkPrice', 'startingBalance', 'startingDate', 'showLeaderboard', 'carryingCapacity', 'degradationRate', 'belowCapacityHealthUpdateStrategy', 'aboveCapacityHealthUpdateStrategy']
     const expectedHeaders = ['Cow Price', 'Milk Price', 'Start Bal', 'Start Date', 'Leaderboard', 'Capacity', 'Degrad Rate', 'BelowCap', 'AboveCap']
 
-    const testId = "ReportHeaderTable"
+    // const testId = "ReportHeaderTable"
```
Comment (katytsao) [inline frontend/src/tests/components/Reports/ReportHeaderTable.test.js]: why not just delete this since it was moved

File: frontend/src/tests/components/Reports/ReportLineTable.test.js [hunk 2]
```diff
@@ -36,7 +38,7 @@ describe("ReportLineTable tests", () => {
 
     expect(screen.getByTestId(`${testId}-cell-row-0-col-userId`)).toHaveTextContent("1");
     expect(screen.getByTestId(`${testId}-cell-row-0-col-username`)).toHaveTextContent("Phill Conrad");
-    expect(screen.getByTestId(`${testId}-cell-row-0-col-totalWealth`)).toHaveTextContent("9745");
+    expect(screen.getByTestId(`${testId}-cell-row-0-col-totalWealth`)).toHaveTextContent("$9,745.00");
     expect(screen.getByTestId(`${testId}-cell-row-0-col-numOfCows`)).toHaveTextContent("3");
     expect(screen.getByTestId(`${testId}-cell-row-0-col-avgCowHealth`)).toHaveTextContent("100");
```
Comment (katytsao) [inline frontend/src/tests/components/Reports/ReportLineTable.test.js:43]: are we okay with 100 health being an integer? would we want a decimal or percentage? (this might be out of scope of this ticket idk)

**Review by katytsao — DISMISSED** [review]
this seems to work fine; might've been easier to just edit css files but i don't really know. also not sure if package-lock is supposed to be edited. but otherwise seems good

File: frontend/package-lock.json
Comment (pconrad) [reply frontend/package-lock.json:1]: Katy is correct: please see: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html

File: frontend/src/tests/components/Reports/ReportHeaderTable.test.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/tests/components/Reports/ReportHeaderTable.test.js]: Agreed with Katy: see: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/tests/components/Reports/ReportLineTable.test.js (hunk 2 above)
Comment (pconrad) [reply frontend/src/tests/components/Reports/ReportLineTable.test.js:43]: Yeah, I think I'd leave that for now.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please see: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html

**Review by pconrad — APPROVED** [review]
LGTM
